canonical_sphinx/config.py

```diff
@@ -160,7 +175,7 @@ def config_inited(_app: Sphinx, config: Any) -> None:  # noqa: ANN401
         ("github_issues", "enabled"),
         ("discourse", "https://discourse.ubuntu.com"),
         ("sequential_nav", "none"),
-        ("display_contributors", True),
+        ("display_contributors", False),
```

Comment:
what is this change?

Reply:
But shouldn't the extension know, at this point, whether the package that has this function was added or not?

Reply:
This feeds HTML context into the design override of Furo. The [HTML template override](https://github.com/canonical/canonical-sphinx/blob/856bd3388de23dcc99424f4c527b40fdde0a45ae/canonical_sphinx/theme/templates/footer.html#L75) has no real way of checking if the function exists - so if this is true by default it will call the function that doesnt exist.

The setup function is affected by the installed pip packages - and that establishes which extensions are loaded, but config_inited has no way of knowing - and for the most part thats fine, nothing will break if you set up an extension that isn't installed and isn't loaded. This seemed cleaner - and easier than passing some value to config_inited, or doing another check and conditionally changing things.

Reply:
But then everybody that wants to display contributors will have to set this to `True`, which used to be the default, right? Isn't that undesirable? I thought the plan was to have this "selective loading" be more transparent to the user - that is, if they install `canonical-sphinx[full]` they get the full behavior with its defaults.

How about this: The extension that provides the `get_contributors_for_file()` function is `canonical.contributor-listing`, right? What if here in `config_inited()` you do

```python
has_contributor_listing = "canonical.contributor-listing" in app.extensions
...
html_context["has_contributor_listing"] = has_contributor_listing
```

... then update the template to read:

```
{% if has_contributor_listing and display_contributors and pagename and page_source_suffix %}
```

Reply:
I've made changes that implement your suggestion here, but I would note that:

* There are roughly as many private, internal documentation sets using the starter pack as public ones, and we do not always want to show contributors for these doc sets (so setting this to default True may not be desirable anyway - and we should actually put some effort into understanding why this default was chosen in the first place).
* The overall design of the contributor listing extension an is pretty poor - in that it relies on a specific template override in our design implementation and wouldn't be usable by other Sphinx users despite being a public extension. With plans to spin off the design overrides into a separate extension, this would mean that you'd need to be loading 3 separate extensions just to have that extension function.

Reply:
I'm fine with changing the defaults of whatever we wish (in my opinion, whatever the documentation team decides to do is fine with me), I'm just against changing a default here, on this unrelated PR. It was also an exercise in using selective dependencies, because I expect that there will possibly be dependency sets other than "minimal" and "full".

No comments on the usability of the contributor listing extension, whatever you decide to do with that is also fine with me :)

Reply:
The way the selecting loading works is that it will load anything on the list that's installed - I felt that was easier to implement than having more granularity in the optional dependencies.

So if you have sphinx-tabs alongside canonical-sphinx in your requirements file, it will be configured.

But you make a reasonable point about changing defaults on an unrelated PR, in retrospect I was just being lazy :P.